compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/FunctionGenerator.kt

```diff
@@ -387,11 +387,9 @@ class FunctionGenerator(declarationGenerator: DeclarationGenerator) : Declaratio
         val constantDefaultValue =
             ConstantExpressionEvaluator.getConstant(valueExpression, context.bindingContext)?.toConstantValue(valueParameterDescriptor.type)
                 ?: error("Constant value expected for default parameter value in annotation, got $valueExpression")
-        return context.irFactory.createExpressionBody(
-            UNDEFINED_OFFSET, UNDEFINED_OFFSET,
-            context.constantValueGenerator.generateConstantValueAsExpression(
-                UNDEFINED_OFFSET, UNDEFINED_OFFSET, constantDefaultValue, valueParameterDescriptor.varargElementType
-            )
-        )
+        val converted = context.constantValueGenerator.generateAnnotationValueAsExpression(
+            UNDEFINED_OFFSET, UNDEFINED_OFFSET, constantDefaultValue, valueParameterDescriptor
+        ) ?: error("Could not convert annotation default $valueExpression")
```

Comment:
`$valueExpression` will always just print the node type, consider adding `getElementTextWithContext()` and/or info about `valueParameterDescriptor`